docs/index.rst

```diff
@@ -9,11 +11,23 @@
 Python 3.5 Filter
 =================
 
-The *fledge-filter-python35* filter allows snippets of Python to be easily written that can be used as filters in Fledge. A similar filter exists that uses Python 2.7 syntax, the |python27| filter. A Python code snippet will be called with sets of asset readings as they or read or processed in a filter pipeline. The data appears in the Python code as a JSON document passed as a Python Dict type.
+The *fledge-filter-python35* filter allows snippets of Python to be easily written that can be used as filters in Fledge. A similar filter exists that uses Python 2.7 syntax, the |python27| filter, however it is recommended that the *python35* filter is used by preference as it provides a more compatible interface to the rest of the Fledge components. 
 
-The user should provide a Python function whose name matches the name given to the plugin when added to the filter pipeline of the south service or north task, e.g. if you name your filter myPython then you should have a function named myPython in the code you enter. This function is send a set of readings to process and should return a set of processed readings. The returned set of readings may be empty if the filter removes all data.
+The purpose of the filter is to allow the user the flexibility to add their own processing into a pipeline that is not supported by any of the existing filters offered by Fledge. The philosophy of Fledge is to provide processing by adding a set of filters that act as a pipeline between a data source and and data sink. The data source may be the south plugin in the south service or the buffered readings data from the storage service in the case of a north service or task. The data sink is either the buffer within the storage service in the case of a south service or the north plugin that sends the data to the upstream system in the case of a north service or task. Each of the filters provides a small, focused operation on the data as it traverses the pipeline, data is passed from one filter in the pipeline to another.
 
-A general code syntax for the function that should be provided is;
+The functionality provided by this filters gives the user the opportunity to write Python code that can manipulate the reading data as it flows, however that modification should follow the same guidelines and principles as followed in the filters that come supplied as part of the Fledge distribution or are contributed by other Fledge users. The overriding principles however are
+
+  - Do not duplicate existing functionality provided by existing filters.
+
+  - Keep the operations small and focused. It is better to have multiple filters each with a specific purpose than to create large, complex Python scripts.
+
+  - Do not buffer large quantities of data, this will effect the footprint of the service and also slow the data pipeline. 
```

Comment:
Overriding guidelines we have already under below section Scripting Guidelines.
We can remove it from here

Reply:
People will probably not read to the end, hence this is here to stress importance.